From: Adam Lackorzynski <adam@l4re.org>
Subject: [PATCH] L4Re: use musl as the libc

L4Re switched from uClibc-ng to musl. musl has the dynamic linker built into
libc, and libstdc++ uses the generic OS configuration with it, as for
linux-musl.

diff -Nurp a/gcc/config/i386/l4re64.h b/gcc/config/i386/l4re64.h
--- a/gcc/config/i386/l4re64.h
+++ b/gcc/config/i386/l4re64.h
@@ -14,7 +14,7 @@
     %{!static: \
       %{!static-pie: \
         %{rdynamic:-export-dynamic} \
-        -dynamic-linker=/rom/libld-l4.so.1}} \
+        -dynamic-linker=/rom/libc.so.1}} \
     %{static:-static} \
     %{static-pie:-static -pie --no-dynamic-linker -z text}}"
 
diff -Nurp a/gcc/config/l4re.h b/gcc/config/l4re.h
--- a/gcc/config/l4re.h
+++ b/gcc/config/l4re.h
@@ -35,7 +35,7 @@
     %{!static: \
       %{!static-pie: \
         %{rdynamic:-export-dynamic} \
-        -dynamic-linker=/rom/libld-l4.so.1}} \
+        -dynamic-linker=/rom/libc.so.1}} \
     %{static:-static} \
     %{static-pie:-static -pie --no-dynamic-linker -z text}}"
 
diff -Nurp a/libstdc++-v3/configure.host b/libstdc++-v3/configure.host
--- a/libstdc++-v3/configure.host
+++ b/libstdc++-v3/configure.host
@@ -256,7 +256,7 @@
     os_include_dir="os/bsd/freebsd"
     ;;
   l4re*)
-    os_include_dir="os/uclibc"
+    os_include_dir="os/generic"
     ;;
   linux-musl*)
     os_include_dir="os/generic"
